src/Data/Primitive/Contiguous/Class.hs

```diff
@@ -1250,6 +1259,20 @@ instance Contiguous (UnliftedArray_ unlifted_a) where
     unsafeFreezeUnliftedArray m
   {-# INLINE run #-}
   run = runUnliftedArrayST
+  {-# INLINE shrink #-}
+  shrink !arr !n = do
+    -- See Note [Shrinking Unlifted Arrays]
+    cloneMutableUnliftedArray arr 0 n
+  {-# INLINE unsafeShrinkAndFreeze #-}
+  unsafeShrinkAndFreeze !arr !n =
+    -- See Note [Shrinking Unlifted Arrays]
+    freezeUnliftedArray arr 0 n
```

Comment:
Won't this be handled by the default `unsafeShrinkAndFreeze`, like for the other 3 array types?

Reply:
The default implementation of `unsafeShrinkAndFreeze` is:

    unsafeShrinkAndFreeze arr0 len' =
      resize arr0 len' >>= unsafeFreeze

Which becomes

    unsafeShrinkAndFreeze arr0 len' =
      resizeUnliftedArray arr0 len' >>= unsafeFreezeUnliftedArray

Where `resizeUnliftedArray` is defined in the shim module and has to consider a case where the array gets larger (which we do not even need defined behavior for in `unsafeShrinkAndFreeze`.

So, this ends up being a bit simpler than the default.